server/libs/domain/src/person/person.service.ts

```diff
@@ -0,0 +1,82 @@
+import { BadRequestException, Inject, Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { AssetResponseDto, mapAsset } from '../asset';
+import { AuthUserDto } from '../auth';
+import { IJobRepository, JobName } from '../job';
+import { ImmichReadStream, IStorageRepository } from '../storage';
+import { PersonUpdateDto } from './dto';
+import { IPersonRepository } from './person.repository';
+import { mapPerson, PersonResponseDto } from './response-dto';
+
+@Injectable()
+export class PersonService {
+  readonly logger = new Logger(PersonService.name);
+
+  constructor(
+    @Inject(IPersonRepository) private repository: IPersonRepository,
+    @Inject(IStorageRepository) private storageRepository: IStorageRepository,
+    @Inject(IJobRepository) private jobRepository: IJobRepository,
+  ) {}
+
+  async getAll(authUser: AuthUserDto): Promise<PersonResponseDto[]> {
+    const people = await this.repository.getAll(authUser.id, { minimumFaceCount: 3 });
```

Comment:
Why a minimum of 3? Just using 1 would be better in my opinion. Would also be useful to have a `limit` option, because on the web we're manually limiting the array to 12

Reply:
This limits what people show up in the explore page. When a person only appears in one photo, it sometimes meant they were a background stranger or a blurry face, not a primary target. Filtering them it until they were in a few photos seemed to improve this selection quite well

Reply:
From that point of view it makes more sense. When testing this feature it was a bit confusing initially, because the person didn't show up on the explore page. I would prefer to see all detected persons and maybe have the option to ignore/hide a person. But we can also see what others think.

Reply:
A few things here:
- Google has _some_ people on the explore page with a "View All" button. We could do the same, based on screen width
- On the all people page, you have the option to show/hide individual faces, we should add this in the future too
- On the all people page, I could not find any detected people with less than 3 associated photos in my own library, so I assumed google did something similar (the definition of a  tagged person is that they appear in at least X photos). Happy to change this in the future is 3 is too high.
- I understand the confusion related to testing a few individual photos. In actual use I would assume that you would not care very much about missing tagged people that are only in 1-2 of your photos. It seems like people you care about tagging would be in 3+, so I'm hoping that's not much of an issue in the "real world".

Reply:
If the intention is to have a page with all detected persons than it isn't an issue. Would perhaps be smart to add a small message with `Only people who occur more than x times are shown` or `It's possible that not all people are shown` to prevent users getting confused and reporting it as an issue.